### DISCUSSION TITLE
include R functions in the model file

### DISCUSSION THREAD

**Discussion by zhg2601** [nlmixr2/nlmixr2#201] [General]
Is it feasible to call in the model file some R functions? For example: 
```r
Q = R_function(V1,V2)
d/[dt](centr) = periph * Q/V2 - centr * (CL/V1 + Q/V1)
```

**Comment by mattfidler** [comment]
Yes, it is a very new feature and is discussed here:

https://nlmixr2.github.io/rxode2/articles/Integrating-User-Defined-Functions-into-rxode2.html

However, I suggest converting the function to `C` if possible which can even create functions for the derivatives as well (and set up the derivative table).  There are currently some stylistic requirements for functions to qualify for that currently.

If you do use the C function, it slows the `nlmixr2` down immensely.  

For now you will have to probably include `nlmixr2` and `rxode2` to use the feature:

```r
library(nlmixr2)
library(rxode2)
```

I am in the process of completing the nlmixr2 submission and will blog about this new feature (and other ones as well).
